crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -520,17 +520,17 @@ mod tests {
     #[derive(Component)]
     struct Parent(Entity);
 
-    impl Traversal for Parent {
-        fn traverse(&self) -> Option<Entity> {
-            Some(self.0)
+    impl<'a> Traversal for &'a Parent {
+        fn traverse<'b>(item: Self::Item<'b>) -> Option<Entity> {
+            Some(item.0)
```

Comment:
These lifetimes can be elided, and same for the trait definition and all the other impls:
```suggestion
    impl Traversal for &'_ Parent {
        fn traverse(item: Self::Item<'_>) -> Option<Entity> {
            Some(item.0)
```